website/content/en/preview/concepts/nodeclaims.md

```diff
@@ -0,0 +1,276 @@
+       "computed new nodeclaim(s) to fit pod(s)","commit":"490ef94","controller":
+       "provisioner","nodeclaims":1,"pods":5}
+    ```
+3. Finds the new NodeClaim and checks its requirements (pre-spin)
+    ```
+    {"level":"INFO","time":"2024-06-22T02:24:16.128Z","logger":"controller","message":   "created nodeclaim","commit":"490ef94","controller":"provisioner","NodePool":
+       {"name":"default"},"NodeClaim":{"name":"default-sfpsl"},"requests":
+       {"cpu":"5150m","pods":"8"},"instance-types":"c3.2xlarge, c4.2xlarge, c4.4xlarge,
+       c5.2xlarge, c5.4xlarge and 55 other(s)"}
+    ```
+4. Based on the NodeClaim’s requirements, directs the cloud provider to create an instance that meets those requirements (launch):
+   Example of log messages at this stage:
+    ```
+    {"level":"INFO","time":"2024-06-22T02:24:19.028Z","logger":"controller","message":   "launched nodeclaim","commit":"490ef94","controller":"nodeclaim.lifecycle",
+       "controllerGroup":"karpenter.sh","controllerKind":"NodeClaim","NodeClaim":
+       {"name":"default-sfpsl"},"namespace":"","name":"default-sfpsl","reconcileID":
+       "9c9dbc80-3f0f-43ab-b01d-faac6c29e979","provider-id":
+       "aws:///us-west-2b/i-08a3bf1cadb205c7e","instance-type":"c3.2xlarge","zone":
+       "us-west-2b","capacity-type":"spot","allocatable":{"cpu":"7910m",
+       "ephemeral-storage":"17Gi","memory":"13215Mi","pods":"58"}}
+    ```
+ 
+5. Gathers necessary metadata from the NodeClaim to check and setup the node.
+6. Checks that the Node exists, and prepares it for use.  This includes:
+    * Waiting for the instance to return a provider ID, instance type, zone, capacity type,
+      and an allocatable status, indicating that the instance is ready.
+    * Checking to see if the node has been synced, adding a finalizer to the node (this provides the same
+      termination guarantees that all Karpenter nodes have), passing in labels, and updating the node owner references.
+      Example of log messages at this stage:
+      ```
+      {"level":"INFO","time":"2024-06-22T02:24:52.642Z","logger":"controller","message":
+         "initialized nodeclaim","commit":"490ef94","controller":
+         "nodeclaim.lifecycle","controllerGroup":"karpenter.sh","controllerKind":
+         "NodeClaim","NodeClaim":{"name":"default-sfpsl"},"namespace":
+         "","name":"default-sfpsl","reconcileID":
+         "7e7a671d-887f-428d-bd79-ddf603290f0a",
+         "provider-id":"aws:///us-west-2b/i-08a3bf1cadb205c7e",
+         "Node":{"name":"ip-192-168-170-220.us-west-2.compute.internal"},
+         "allocatable":{"cpu":"7910m","ephemeral-storage":"18242267924",
+         "hugepages-2Mi":"0","memory":"14320468Ki","pods":"58"}}
+      ```
+7. Making sure that the Node is ready to use. This includes such things as seeing if resources are registered and start-up taints are removed.
+8. Checking the nodes for liveliness.
```

Comment:
This isn't really a single step, but happens periodically for the node's lifetime. I'm not sure it really fits here.